src/releaseNotes.ts

```diff
@@ -0,0 +1,293 @@
+
+  try {
+    await exec('ditto', ['-xk', appPath, workingDir], {silent: true})
+
+    const payloadDirectory = join(workingDir, 'Payload')
+    const entries = await readdir(payloadDirectory)
+    const appDirectory = entries.find(entry => entry.endsWith('.app'))
+
+    if (!appDirectory) {
+      throw new Error(
+        'Unable to locate *.app bundle inside TestFlight payload.'
+      )
+    }
+
+    const infoPath = join(payloadDirectory, appDirectory, 'Info.plist')
+    const infoJson = await readPlistAsJson(infoPath)
+    const parsed = JSON.parse(infoJson) as Record<string, string>
+
+    const bundleId = parsed['CFBundleIdentifier']
+    const buildNumber = parsed['CFBundleVersion']
+
+    if (!bundleId || !buildNumber) {
+      throw new Error(
+        'Info.plist missing CFBundleIdentifier or CFBundleVersion.'
+      )
+    }
+
+    return {bundleId, buildNumber}
+  } finally {
+    await rmRF(workingDir)
+  }
+}
+
+async function readPlistAsJson(plistPath: string): Promise<string> {
+  let output = ''
+  await exec('plutil', ['-convert', 'json', '-o', '-', plistPath], {
+    silent: true,
+    listeners: {
+      stdout: (data: Buffer) => {
+        output += data.toString()
+      }
+    }
+  })
+
+  return output
+}
+
+async function lookupAppId(bundleId: string, token: string): Promise<string> {
+  const params = new URLSearchParams()
+  params.set('filter[bundleId]', bundleId)
+
+  const response = await fetchJson(
+    `/apps?${params.toString()}`,
+    token,
+    'Failed to locate App Store Connect application.'
+  )
+
+  const appId = response.data?.[0]?.id
+  if (!appId) {
+    throw new Error(
+      `Unable to find App Store Connect app for bundle id ${bundleId}.`
+    )
+  }
+
+  return appId
+}
+
+async function lookupBuildId(
+  appId: string,
+  buildNumber: string,
+  platform: string,
+  token: string
+): Promise<string> {
+  const params = new URLSearchParams()
+  params.set('filter[app]', appId)
+  params.set('filter[version]', buildNumber)
+  params.set('filter[preReleaseVersion.platform]', platform)
+
+  for (let attempt = 0; attempt < MAX_ATTEMPTS; attempt++) {
+    const response = await fetchJson(
+      `/builds?${params.toString()}`,
+      token,
+      'Failed to query builds for release note update.'
+    )
+
+    const buildId = response.data?.[0]?.id
+    if (buildId) {
+      return buildId
+    }
+
+    warning(
+      `Build ${buildNumber} not yet visible in App Store Connect (attempt ${
+        attempt + 1
+      }/${MAX_ATTEMPTS}). Retrying in ${Math.round(RETRY_DELAY_MS / 1000)}s`
+    )
+    await delay(RETRY_DELAY_MS)
+  }
+
+  throw new Error(
+    `Timed out waiting for build ${buildNumber} to appear in App Store Connect.`
+  )
+}
+
+async function lookupLocalizationId(
+  buildId: string,
+  token: string
+): Promise<string> {
+  for (let attempt = 0; attempt < MAX_ATTEMPTS; attempt++) {
+    const response = await fetchJson(
+      `/builds/${buildId}/betaBuildLocalizations`,
```

Comment:
`beta` is just referring to the fact that it's a TestFlight build?

Reply:
Correct. Here's the [full documentation](https://developer.apple.com/documentation/appstoreconnectapi) for the reference.

> [!NOTE] 
> A `betaBuildLocalizations` resource represents the localized content that appears in the “What’s New” text in TestFlight. You should change this text for every build.
> https://developer.apple.com/documentation/appstoreconnectapi/beta-build-localizations